utilities/download_old_firefox.py

```diff
@@ -9,43 +9,19 @@
     today = date.today()
     # Set correct month for URL builder
     if len(f"{today.month}") < 2:
-        current_month = f"0{today.month}"
+        download_month = f"0{today.month - 1}"
     else:
-        current_month = f"{today.month}"
+        download_month = f"{today.month - 1}"
 
-    # Set correct month for URL builder
-    if len(f"{today.day}") < 2 and int(today.day - 5) < 0:
-        download_day = "01"
-    elif int(today.day) < 15:
-        download_day = f"0{today.day - 5}"
-    else:
-        download_day = f"{today.day - 5}"
-
-    download_month = f"{current_month}"
-
-    # if its a new month just grab the 28th day build of last month and build URL
-    if int(download_day) < 6:
-        if int(current_month) < 10:
-            download_month = f"0{int(current_month) - 1}"
-        else:
-             download_month = f"{int(current_month) - 1}"
-        download_dir = f"{base_url}/pub/firefox/nightly/{today.year}/{download_month}/"
-        html = requests.get(download_dir)
+    download_dir = f"{base_url}/pub/firefox/nightly/{today.year}/{download_month}/"
+    html = requests.get(download_dir)
 
-        soup = BeautifulSoup(html.text, "html.parser")
-        page_link = soup.find_all(
-            href=re.compile(f"{today.year}-{download_month}-28.*-mozilla-central")
-        )
-    else:
-        download_dir = f"{base_url}/pub/firefox/nightly/{today.year}/{download_month}/"
-        html = requests.get(download_dir)
-
-        soup = BeautifulSoup(html.text, "html.parser")
-        page_link = soup.find_all(
-            href=re.compile(
-                f"{today.year}-{download_month}-{download_day}.*-mozilla-central"
-            )
+    soup = BeautifulSoup(html.text, "html.parser")
+    page_link = soup.find_all(
+        href=re.compile(
+            f"{today.year}-{download_month}-15.*-mozilla-central"
```

Comment:
I'm too lazy to check, but... are we sure that each month has a build on the 15<sup>th</sup>; or is it possible we might not have a build if that happens to be a Sunday or something?

Reply:
Hhmm, I tried rebuilding in Node.js but it interestingly didn't pick up the -l10n folders like I expected:
```js
// import * as cheerio from "cheerio";

for (let month = 1; month <= 12; month += 1) {
  const matches = await fetchDLPage(2024, month);
  for (const [str] of matches) {
    console.log(str);
  }
}

async function fetchDLPage(year, month) {
  const _month = month.toString().padStart(2, "0");
  const re = new RegExp(`${year}-${_month}-15.*-mozilla-central`);
  const url = `https://download-installer.cdn.mozilla.net/pub/firefox/nightly/${year}/${_month}/`;
  const html = await fetch(url).then((r) => r.text());
  return html.matchAll(re.toString());
}
```

Although my RegExp's did match more than expected:
```sh
node index

/2024-01-15-09-46-19-mozilla-central/">2024-01-15-09-46-19-mozilla-central/
/2024-01-15-20-39-16-mozilla-central/">2024-01-15-20-39-16-mozilla-central/
/2024-02-15-09-55-51-mozilla-central/">2024-02-15-09-55-51-mozilla-central/
/2024-02-15-21-01-02-mozilla-central/">2024-02-15-21-01-02-mozilla-central/
/2024-03-15-09-43-32-mozilla-central/">2024-03-15-09-43-32-mozilla-central/
/2024-03-15-20-58-19-mozilla-central/">2024-03-15-20-58-19-mozilla-central/
/2024-04-15-08-26-38-mozilla-central/">2024-04-15-08-26-38-mozilla-central/
/2024-04-15-20-39-32-mozilla-central/">2024-04-15-20-39-32-mozilla-central/
/2024-05-15-03-23-56-mozilla-central/">2024-05-15-03-23-56-mozilla-central/
/2024-05-15-20-24-18-mozilla-central/">2024-05-15-20-24-18-mozilla-central/
/2024-06-15-09-27-45-mozilla-central/">2024-06-15-09-27-45-mozilla-central/
/2024-06-15-21-31-13-mozilla-central/">2024-06-15-21-31-13-mozilla-central/
/2024-07-15-09-51-03-mozilla-central/">2024-07-15-09-51-03-mozilla-central/
/2024-07-15-21-15-55-mozilla-central/">2024-07-15-21-15-55-mozilla-central/
/2024-08-15-09-36-22-mozilla-central/">2024-08-15-09-36-22-mozilla-central/
/2024-08-15-21-31-44-mozilla-central/">2024-08-15-21-31-44-mozilla-central/
/2024-09-15-21-50-47-mozilla-central/">2024-09-15-21-50-47-mozilla-central/
/2024-10-15-09-32-00-mozilla-central/">2024-10-15-09-32-00-mozilla-central/
/2024-10-15-13-57-09-mozilla-central/">2024-10-15-13-57-09-mozilla-central/
/2024-10-15-21-41-57-mozilla-central/">2024-10-15-21-41-57-mozilla-central/
/2024-11-15-09-34-47-mozilla-central/">2024-11-15-09-34-47-mozilla-central/
/2024-11-15-21-07-54-mozilla-central/">2024-11-15-21-07-54-mozilla-central/
/2024-12-15-09-08-17-mozilla-central/">2024-12-15-09-08-17-mozilla-central/
/2024-12-15-21-41-01-mozilla-central/">2024-12-15-21-41-01-mozilla-central/
```

Reply:
Not 100% sure but it is a nightly so there should be a build unless something failed. These images aren't built that often though.